apps/confidential/confidential.go

```diff
@@ -0,0 +1,364 @@
+during creation. As it doesn't have any pointers in it, anything borrowed from it, such as
+Base.AuthParams is a copy that is free to be manipulated here.
+
+Duplicate Calls shared between public.Client and this package:
+There is some duplicate call options provided here that are the same as in public.Client . This
+is a design choices. Go proverb(https://www.youtube.com/watch?v=PAAkCSZUG1c&t=9m28s):
+"a little copying is better than a little dependency". Yes, we could have another package with
+shared options (fail).  That divides like 2 options from all others which makes the user look
+through more docs.  We can have all clients in one package, but I think separate packages
+here makes for better naming (public.Client vs client.PublicClient).  So I chose a little
+duplication.
+
+.Net People, Take note on X509:
+This uses x509.Certificates and private keys. x509 does not store private keys. .Net
+has some x509.Certificate2 thing that has private keys, but that is just some bullcrap that .Net
+added, it doesn't exist in real life.  Seriously, "x509.Certificate2", bahahahaha.  As such I've
+put a PEM decoder into here.
+*/
+
+// TODO(msal): This should have example code for each method on client using Go's example doc framework.
+// base usage details should be includee in the package documentation.
+
+// AuthenticationResult contains the results of one token acquisition operation.
+// For details see https://aka.ms/msal-net-authenticationresult
+type AuthenticationResult = base.AuthenticationResult
+
+type Account = shared.Account
+
+// CertFromPEM converts a PEM file (.pem or .key) for use with NewCredFromCert(). The file
+// must have the public certificate and the private key encoded. The private key must be encoded
+// in PKCS8 (not PKCS1). This is usally denoted by the section "PRIVATE KEY" (instead of PKCS1's
+// "RSA PRIVATE KEY"). If a PEM block is encoded and password is not an empty string, it attempts
+// to decrypt the PEM blocks using the password. This will return multiple x509 certificates,
+// though this use case should have a single cert. Multiple certs are due to certificate
+// chaining for use cases like TLS that sign from root to leaf.
+func CertFromPEM(pemData []byte, password string) ([]*x509.Certificate, crypto.PrivateKey, error) {
+	var certs []*x509.Certificate
+	var priv crypto.PrivateKey
+	for {
+		block, rest := pem.Decode(pemData)
+		if block == nil {
+			break
+		}
+
+		if x509.IsEncryptedPEMBlock(block) {
+			if password != "" {
+				b, err := x509.DecryptPEMBlock(block, []byte(password))
+				if err != nil {
+					return nil, nil, fmt.Errorf("could not decrypt encrypted PEM block: %w", err)
+				}
+				block, _ = pem.Decode(b)
+				if block == nil {
+					return nil, nil, fmt.Errorf("encounter encrypted PEM block that did not decode")
+				}
+			} else {
+				return nil, nil, fmt.Errorf("encountered encrypted PEM block, but password was not passed")
```

Comment:
>return nil, nil, fmt.Errorf("encountered encrypted PEM block, but password was not passed") [](start = 4, length = 91)

in Python and Java I believe we recently had a situation where the password was blank and that is a valid scenario. not sure if that is the same case here but it seems related.

Reply:
Since we think that is a valid scenario, I fixed this to try with the password if the string is the zero value.